coriolis/osmorphing/debian.py

```diff
@@ -122,6 +122,14 @@ def set_net_config(self, nics_info, dhcp):
             cfg_name = "%s/coriolis_netplan.yaml" % netplan_base
             self._write_file_sudo(cfg_name, new_cfg)
 
+    def check_installed_package(self, package_name):
+        cmd = 'dpkg-query -W %s' % ("".join(package_name))
```

Comment:
I think it would be better to just send a single command that lists all the packages installed on the migrated system. Parse that list and save it on the tools object. Then when you need to check a package, you just do a `if package in self.installed_packages: return` or do a set difference like below to get the list of packages that need to be installed. 

The point of this is to send a minimum amount of SSH commands, instead of sending N+1commands for each package we want to install. For Debian-based, you can use `apt list --installed`, please check a command for the others too.

Reply:
You also don't need to `"".join(package_name)` here, you can simply pass the string value to the string format.